### PR TITLE
Feature/issue 87

### PR DESCRIPTION
**Pull request by mkryva**

### CODE REVIEW

File: tests/unit/destination/ssh/test_write_status.py [hunk 1]
```diff
@@ -6,19 +6,19 @@
 from twindb_backup.destination.ssh import Ssh
 from twindb_backup.ssh.client import SshClient
```
Comment (akuzminsky) [inline tests/unit/destination/ssh/test_write_status.py]: If the test is not needed, delete it.

File: tests/unit/destination/test_base.py [hunk 2]
```diff
@@ -8,14 +8,14 @@
 
 
 # noinspection PyUnresolvedReferences
+#@mock.patch('twindb_backup.destination.base_destination.json')
```
Comment (akuzminsky) [inline tests/unit/destination/test_base.py]: Again, keep the code clean.

File: tests/unit/destination/test_base.py [hunk 3]
```diff
@@ -8,14 +8,14 @@
...
 @mock.patch.object(BaseDestination, '_write_status')
 @mock.patch.object(BaseDestination, '_read_status')
 def test_status(mock_read_status, mock_write_status):
     dst = BaseDestination()
-    dst.status(status='foo')
-    mock_write_status.assesrt_called_once_with('foo')
+    dst.status(status={})
+    mock_write_status.assesrt_called_once_with({})
     dst.status()
-    mock_read_status.assert_called_once_with()
```
Comment (akuzminsky) [inline tests/unit/destination/test_base.py]: This test needs to be rewritten. It doesn't test what the method does.

File: twindb_backup/destination/base_destination.py [hunk 4]
```diff
@@ -115,7 +119,13 @@ def status(self, status=None):
         :return: dictionary with the status
         """
         if status:
-            return self._write_status(status)
+            status['checksum'] = hashlib.md5(
```
Comment (akuzminsky) [inline twindb_backup/destination/base_destination.py]: The method adds a key to the dictionary. We should better describe the status format.
Also, I see `status` type is not specified in the docstring - it should be.

File: twindb_backup/destination/base_destination.py [hunk 5]
```diff
@@ -125,7 +135,18 @@ def _write_status(self, status):
 
     @abstractmethod
     def _read_status(self):
-        """Function that actually reads status"""
+        pass
+
+    def _read_status_with_safe(self, status_path, status_tmp_path):
```
Comment (akuzminsky) [inline twindb_backup/destination/base_destination.py]: Too many nested levels - it's non-testable.

File: twindb_backup/destination/local.py [hunk 6]
```diff
@@ -32,6 +31,10 @@ def __init__(self, path=None):
             path=self.path,
             hostname=socket.gethostname()
         )
+        self.status_tmp_path = "{path}/{hostname}/status.tmp".format(
```
Comment (akuzminsky) [inline twindb_backup/destination/local.py]: Make it simpler
```
self.status_tmp_path = self.status_path + ".tmp"
```

File: twindb_backup/destination/local.py [hunk 7]
```diff
@@ -82,18 +85,24 @@ def get_stream(path):
...
+            self._move_file(self.status_tmp_path, self.status_path)
+            return
+        raise StatusFileError("Valid status file not found")
 
-    def _read_status(self):
-        if not self._status_exists():
-            return self._empty_status
+    def _get_file_content(self, path):
+        f_descr = open(path, "r")
+        return f_descr.read()
```
Comment (akuzminsky) [inline twindb_backup/destination/local.py]: Let's make it nicer, and not leak a file descriptor
```
with open(path) as f_descr:
    return f_descr.read()
```

File: twindb_backup/destination/local.py [hunk 8]
```diff
@@ -82,18 +85,24 @@ def get_stream(path):
...
+
+    def _move_file(self, source, destination):
+        cmd = ["cp", "-rf", source, destination]
+        run_command(cmd)
 
-        with open(self.status_path) as status_descriptor:
-            cout = status_descriptor.read()
-            return json.loads(base64.b64decode(cout))
+    def _read_status(self):
+        return self._read_status_with_safe(self.status_path,
```
Comment (akuzminsky) [inline twindb_backup/destination/local.py]: Do you really need to pass these arguments? They're the class' attributes.

**Comment by akuzminsky** [comment]
ah, and the integration tests fail

File: twindb_backup/destination/base_destination.py (hunk 4 above)
Comment (mkryva) [reply twindb_backup/destination/base_destination.py]: Checksum is only for internal use, not for public. I've noticed it in code + updated docstring.

**Pull request merged**